File: 백준/Bronze/31962. 등교/등교.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n,x=map(int,input().split())
maxn=-1
for _ in range(n):
    s,t=map(int,input().split())
    if s+t<=x:
        if maxn<s:
            maxn=s;
print(maxn)

File: 백준/Bronze/31962. 등교/등교.cc
#include <iostream>
using namespace std;
int main(void) {
    int n, x, s, t, maxn = -1;
    cin >> n >> x;
    for (int i = 0; i < n; i++) {
        cin >> s >> t;
        if (s + t <= x)
            if (maxn < s)
                maxn = s;
    }
    cout << maxn;
    return 0;
}